docs/cpu-constraints.md

```diff
@@ -0,0 +1,28 @@
+# CPU Constraints in Clear Containers
+
+## Default number of virtual CPUs
+
+Before starting a container, the runtime reads the option `default_vcpus` from configuration file to know the number of virtual CPUs (vCPUs) to start the virtual machine. In order to have fast boot time and small memory footprint per virtual machine, by default `default_vcpus` is equal to 1, be aware that increasing this value will impact negatively virtual machine's boot time and memory footprint.
+The recommendation is to do not touch this variable unless you know what are you doing. If your container needs more than 1 vCPU you can use [docker `--cpus`][1] or [kubernetes `cpu` limits][2] to assign more resources, the runtime will hot plug in the virtual machine the number of vCPUs required by the container and the agent will place the container inside a new CPU cgroup.
+
+## Virtual CPUs and Kubernetes PODs
+
+A kubernetes POD is a group of one or more containers, with shared storage/network, and a specification for how to run the containers [[1][3]]. In Clear Containers this group of containers run inside the same virtual machine, when a new container is created with a CPU constraint, using CPU quota and period the runtime calculates how many vCPUs are required by the container, the formula to calculate this is: `(quota + (period - 1)) / period`. If a CPU constraint is not specified, the runtime will not hot add more vCPUs and the container will not be placed inside a CPU cgroup, it will use the number of vCPUs specified by `default_vcpus` and share these resources with other containers in the same situation (without CPU constraint).
+
+
+## Container life cycle
+
+As was mention above, when a container is created with a CPU constraint, the runtime will hot add the number of vCPUs required by the container, in the same way, when it is stopped the runtime will hot remove these resources.
```

Comment:
Suggested rewrite: When you create a container with a CPU constraint, the runtime hot adds the number of vCPUs required by the container. Similarly, when the container stops, the runtime hot removes these resources.